src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/AppealReferenceNumberHandler.java

```diff
@@ -32,31 +32,31 @@ public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCa
         requireNonNull(callback, "callback must not be null");
 
         return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
-                && callback.getEvent() == Event.SUBMIT_APPEAL;
+               && callback.getEvent() == Event.SUBMIT_APPEAL;
     }
 
     @Override
     public PreSubmitCallbackResponse<AsylumCase> handle(
-            PreSubmitCallbackStage callbackStage,
-            Callback<AsylumCase> callback) {
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback) {
 
         if (!canHandle(callbackStage, callback)) {
             throw new IllegalStateException("Cannot handle callback");
         }
 
         AsylumCase asylumCase = callback
-                .getCaseDetails()
-                .getCaseData();
+            .getCaseDetails()
+            .getCaseData();
 
         PreSubmitCallbackResponse<AsylumCase> callbackResponse =
-                new PreSubmitCallbackResponse<>(asylumCase);
+            new PreSubmitCallbackResponse<>(asylumCase);
 
         if (!asylumCase.getAppealReferenceNumber().isPresent()) {
 
             Optional<String> appealReferenceNumber =
                 appealReferenceNumberGenerator.getNextAppealReferenceNumberFor(
-                        asylumCase.getAppealType()
-                            .orElseThrow(() -> new AsylumCaseRetrievalException("Unrecognised asylum case type")));
+                    asylumCase.getAppealType()
+                        .orElseThrow(() -> new AppealReferenceNumberInitializerException("Unrecognised asylum case type")));
 
```

Comment:
I'll have to take another look at my formatting, I did import the settings you sent me but obviously something still a miss